circularqueue.gen.go.tmpl

```diff
@@ -0,0 +1,114 @@
+//lint:file-ignore U1000 this is generated code
+package kapacitor
+
+{{with $types := .}}{{range $k := $types}}
+
+// {{ $k }}CircularQueue defines a circular queue, always use the contructor to create one.
+type {{ $k }}CircularQueue struct {
+	data     []{{ $k }}
+	head     int
+	tail     int
```

Comment:
invariant: `tail = (head + l) % len(data)` right? Is it worth keeping it separately?

Reply:
Yes, its mostly so we don't have to do a mod over and over again.